api/python/tests/test_formats.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/python
```

Comment:
Still not fixed?

Reply:
:-(  I removed that..  ..ah.  I thought your comment was in 'formats.py'.

Reply:
> So anyways... Until we switch to proper URI classes, can you add the unquote calls?

Definitely.

Reply:
> Can you refactor this?

..hokay.

Reply:
> Refactor

To be clear, we'll still probably want to do *something* to enable creation of simple formats by users.  That was made that way so that one call could create a format in cases where serialization / deserialization isn't a complex matter at all (JSON, Unicode, yaml, and similar formats that don't have a lot of underlying complexity and variance like CSV and some others do.  ..you have any complaints about a class factory function, then?